### PR TITLE
Support installing translations for all installed plugins/themes

### PR DESCRIPTION
**Pull request by ocean90**
See #59.

### CODE REVIEW

**Comment by ocean90** [comment]
@schlessera provided some feedback on Slack for this: https://wordpress.slack.com/archives/C02RP4T41/p1539759949000100

tldr; Consider using a table for the output which should make it easier to parse the result of each installation.

**Comment by ocean90** [comment]
Current output is inspired by `wp plugin update --all`:

<details>
<summary>wp language plugin install --all de_DE de_CH</summary>

```
Warning: Language 'de_DE' not available.
Language 'de_DE' for 'Airplane Mode' not installed.
Warning: Language 'de_CH' not available.
Language 'de_CH' for 'Airplane Mode' not installed.
Downloading translation from https://downloads.wordpress.org/translation/plugin/akismet/4.0.3/de_DE.zip...
Unpacking the update...
Installing the latest version...
Translation updated successfully.
Language 'de_DE' for 'Akismet Anti-Spam' installed.
Downloading translation from https://downloads.wordpress.org/translation/plugin/akismet/3.1.11/de_CH.zip...
Unpacking the update...
Installing the latest version...
Translation updated successfully.
Language 'de_CH' for 'Akismet Anti-Spam' installed.
Language 'de_DE' for 'Classic Editor' already installed.
....
Warning: Language 'de_CH' not available.
Language 'de_CH' for 'WP Team List' not installed.
Language 'de_DE' for 'WP Team List' already installed.
Language 'de_CH' for 'WP Team List' already installed.
Language 'de_DE' for 'WP Widget Disable' already installed.
Language 'de_CH' for 'WP Widget Disable' already installed.
+---------------------------------+--------+-------------------+
| name                            | locale | status            |
+---------------------------------+--------+-------------------+
| airplane-mode-master            | de_DE  | not available     |
| airplane-mode-master            | de_CH  | not available     |
| akismet                         | de_DE  | installed         |
| akismet                         | de_CH  | installed         |
| classic-editor                  | de_DE  | already installed |
| classic-editor                  | de_CH  | not available     |
....
| rplus-wp-team-list-git          | de_DE  | not available     |
| rplus-wp-team-list-git          | de_CH  | not available     |
| wp-team-list                    | de_DE  | already installed |
| wp-team-list                    | de_CH  | already installed |
| wp-widget-disable               | de_DE  | already installed |
| wp-widget-disable               | de_CH  | already installed |
+---------------------------------+--------+-------------------+
Success: Installed 2 of 86 languages (84 skipped).
```
</details>

<details>
<summary>wp language plugin install --all de_DE --format=summary</summary>

```
Warning: Language 'de_DE' not available.
Language 'de_DE' for 'Airplane Mode' not installed.
Language 'de_DE' for 'Akismet Anti-Spam' already installed.
Language 'de_DE' for 'Classic Editor' already installed.
Warning: Language 'de_DE' not available.
Language 'de_DE' for 'Core Control' not installed.
Warning: Language 'de_DE' not available.
Language 'de_DE' for 'CSS' not installed.
....
Language 'de_DE' for 'WP Team List' not installed.
Language 'de_DE' for 'WP Team List' already installed.
Language 'de_DE' for 'WP Widget Disable' already installed.
Success: Installed 0 of 43 languages (43 skipped).
```
</details>

<details>
<summary>wp language plugin install --all de_DE --format=json</summary>

```
[{"name":"airplane-mode-master","locale":"de_DE","status":"not available"},{"name":"akismet","locale":"de_DE","status":"already installed"},{"name":"classic-editor","locale":"de_DE","status":"already installed"},{"name":"core-control","locale":"de_DE","status":"not available"},{"name":"css","locale":"de_DE","status":"not available"},{"name":"debug-bar","locale":"de_DE","status":"not available"},{"name":"debug-bar-console","locale":"de_DE","status":"not available"},{"name":"debug-bar-transients","locale":"de_DE","status":"already installed"},{"name":"disable-privacy-tools","locale":"de_DE","status":"already installed"},{"name":"drafts-of-post-revisions","locale":"de_DE","status":"not available"},{"name":"dynamic-hostname","locale":"de_DE","status":"not available"},{"name":"genericond","locale":"de_DE","status":"not available"},{"name":"gutenberg-git","locale":"de_DE","status":"not available"},{"name":"gutenberg","locale":"de_DE","status":"already installed"},{"name":"gutenberg-testing","locale":"de_DE","status":"not available"},{"name":"test-plugin","locale":"de_DE","status":"not available"},{"name":"h2push","locale":"de_DE","status":"not available"},{"name":"members","locale":"de_DE","status":"already installed"},{"name":"ms-global-media","locale":"de_DE","status":"not available"},{"name":"o2-git","locale":"de_DE","status":"not available"},{"name":"parent-term-toggler","locale":"de_DE","status":"not available"},{"name":"polylang","locale":"de_DE","status":"already installed"},{"name":"preserve-editor-scroll-position","locale":"de_DE","status":"not available"},{"name":"public-post-preview-git","locale":"de_DE","status":"not available"},{"name":"public-post-preview","locale":"de_DE","status":"already installed"},{"name":"query-monitor","locale":"de_DE","status":"already installed"},{"name":"relax-comment-filters","locale":"de_DE","status":"not available"},{"name":"rest-post-likes","locale":"de_DE","status":"not available"},{"name":"rewrite-rules-inspector","locale":"de_DE","status":"not available"},{"name":"shiny-updates","locale":"de_DE","status":"not available"},{"name":"shortcode-ui","locale":"de_DE","status":"already installed"},{"name":"slack","locale":"de_DE","status":"not available"},{"name":"solr-power","locale":"de_DE","status":"not available"},{"name":"user-switching","locale":"de_DE","status":"already installed"},{"name":"woocommerce","locale":"de_DE","status":"already installed"},{"name":"wordpress-importer","locale":"de_DE","status":"already installed"},{"name":"wordpress-importer-git","locale":"de_DE","status":"not available"},{"name":"wp-feed-post-thumbnail","locale":"de_DE","status":"already installed"},{"name":"rest-api","locale":"de_DE","status":"not available"},{"name":"rest-api-oauth1","locale":"de_DE","status":"not available"},{"name":"rplus-wp-team-list-git","locale":"de_DE","status":"not available"},{"name":"wp-team-list","locale":"de_DE","status":"already installed"},{"name":"wp-widget-disable","locale":"de_DE","status":"already installed"}]%
```
</details>

<br>

Let me know what you think. If it's good I'm going to do the same for themes and add some tests.

**Comment by ocean90** [comment]
(The `command:language-core-install` label should probably be removed as the PR only touches plugins and themes.)

**Comment by schlessera** [comment]
@ocean90 What do you think about combining these two in the "live status messages" (not the table):
```
Warning: Language 'de_CH' not available.
Language 'de_CH' for 'WP Team List' not installed.
```
This would then turn into this instead:
```
Warning: Language 'de_CH' for 'WP Team List' not available.
```
This already implies that it will not be installed.

Would this make it clearer, or does it worsen overall consistency?

**Comment by schlessera** [comment]
Otherwise, I like the current version and think this is good to go.

**Comment by ocean90** [comment]
> What do you think about combining these two in the "live status messages" 

The output next to "Warning" is just one of the `WP_Error` messages which [download_language_pack()](https://github.com/wp-cli/language-command/blob/a7c45c9a14885a1510f488727e66f76babb395e8/src/WP_CLI/CommandWithTranslation.php#L209) can return. We could extend the changes from #69 to also adjust the logging but not sure if it's really worth since it would only cover one case. Maybe worth it's own issue/PR.

File: features/language-plugin.feature [hunk 1]
```diff
@@ -323,3 +323,51 @@ Feature: Manage translation files for a WordPress install
...
+    And I run `wp plugin install akismet --version=4.0 --force`
+    And I run `wp plugin install jetpack --version=6.4 --force`
+
+    When I run `wp language plugin install de_DE`
+    Then the return code should be 1
+    And STDERR should be:
+      """
+      Error: Please specify a plugin, or use --all.
+      """
+    And STDOUT should be empty
```
Comment (ocean90) [inline features/language-plugin.feature]: The test is failing but I'm not sure why. Same for the theme command.

Comment (schlessera) [reply features/language-plugin.feature]: @ocean90 For a command you expect to fail, you need to use `When I try ...` not `When I run ...`.

Comment (ocean90) [reply features/language-plugin.feature]: 🤦‍♂️ Thanks!

File: features/language-plugin.feature [hunk 2]
```diff
@@ -323,3 +323,51 @@ Feature: Manage translation files for a WordPress install
       | akismet  | de_DE    | available | installed |
       | jetpack  | de_DE    | none      | installed |
     And STDERR should be empty
+
+
+  @require-wp-4.0
+  Scenario: Install translations for all installed plugin
```
Comment (swissspidy) [inline features/language-plugin.feature]:
```suggestion
  Scenario: Install translations for all installed plugins
```

File: src/Plugin_Language_Command.php [hunk 3]
```diff
@@ -244,6 +278,82 @@ public function install( $args, $assoc_args ) {
 		\WP_CLI\Utils\report_batch_operation_results( 'language', 'install', $count, $successes, $errors, $skips );
 	}
 
+	/**
+	 * Installs translations for all installed plugin.
```
Comment (swissspidy) [inline src/Plugin_Language_Command.php]:
```suggestion
	 * Installs translations for all installed plugins.
```

**Comment by strarsis** [comment]
Does this work in latest release of WP CLI (2.2.0) (and WordPress (5.1.1))?
`wp core language update` states `Success: Translations are up to date.` 
but all plugin translations are still missing, only core translations are installed.

**Comment by strarsis** [comment]
This doesn't work when using an alias:
````
wp @staging language plugin install --all de_DE
````

````
usage: wp language plugin install <plugin> <language>...
````

**Pull request merged**